cas-server-webapp/src/main/webapp/WEB-INF/webflow/login/login-webflow.xml

```diff
@@ -101,11 +101,12 @@
 
     <action-state id="beforeLogin">
         <evaluate expression="loginHandler.beforeLogin(flowRequestContext)" />
-        <transition on="osf" to="viewLoginForm" />
-        <transition on="institution" to="institutionLogin" />
+        <transition on="osfDefaultLogin" to="viewLoginForm" />
+        <transition on="osfCampaignLogin" to="viewLoginForm" />
```

Comment:
Currently,  both `osfDefaultLogin` and `osfCampaignLogin` lands on `viewLoginForm`. Explicitly separating them in login flow makes the logic clear and future-proof.